server/src/main/java/io/kroki/server/service/Structurizr.java

```diff
@@ -89,10 +97,23 @@ public void convert(String sourceDecoded, String serviceName, FileFormat fileFor
         // take the first view if not specified
         selectedView = views.iterator().next();
       }
-      StringWriter stringWriter = new StringWriter();
-      structurizrPlantUMLWriter.write(selectedView, stringWriter);
-      String plantumlDiagram = stringWriter.toString();
-      return Plantuml.convert(plantumlDiagram, fileFormat, new JsonObject());
+      final Diagram diagram;
+      if (selectedView instanceof DynamicView) {
+        diagram = structurizrPlantUMLExporter.export((DynamicView) selectedView);
+      } else if (selectedView instanceof DeploymentView) {
+        diagram = structurizrPlantUMLExporter.export((DeploymentView) selectedView);
+      } else if (selectedView instanceof ComponentView) {
+        diagram = structurizrPlantUMLExporter.export((ComponentView) selectedView);
+      } else if (selectedView instanceof ContainerView) {
+        diagram = structurizrPlantUMLExporter.export((ContainerView) selectedView);
+      } else if (selectedView instanceof SystemContextView) {
+        diagram = structurizrPlantUMLExporter.export((SystemContextView) selectedView);
+      } else if (selectedView instanceof SystemLandscapeView) {
+        diagram = structurizrPlantUMLExporter.export((SystemLandscapeView) selectedView);
+      } else {
+        throw new BadRequestException("View type is not supported: " + selectedView.getClass().getSimpleName() + ", must be a DynamicView, DeploymentView, ComponentView, ContainerView, SystemContextView or SystemLandscapeView.");
+      }
```

Comment:
@simonbrowndotje Thanks for your guidance, I've replaced structurizr-plantuml in favour of structurizr-export. Not sure if this is the best way to export a `View` to a `Diagram`?

Alternatively, I could export the whole `Workspace` and then filter by view id but I suppose it will be less performant.